UI/Panels/Device/MFShiftRegisterPanel.cs

```diff
@@ -29,52 +25,59 @@ public MFShiftRegisterPanel()
 
         public MFShiftRegisterPanel(ShiftRegister shiftRegister, List<MobiFlightPin> Pins) : this()
         {
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin1ComboBox, Pins, shiftRegister.LatchPin);
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin2ComboBox, Pins, shiftRegister.ClockPin);
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin3ComboBox, Pins, shiftRegister.DataPin);
+            pinList = Pins; // Keep pin list stored
 
-            if (mfPin1ComboBox.Items.Count > 2)
-            {
-                mfPin1ComboBox.SelectedIndex = 0;
-                mfPin2ComboBox.SelectedIndex = 1;
-                mfPin3ComboBox.SelectedIndex = 2;                
-            }
+            this.shiftRegister = shiftRegister;
+            update_lists();
```

Comment:
Let's change this to a more speaking name and use camel case:
 UpdateFreePinsInDropDowns()